apps/osvr_server.cpp

```diff
@@ -50,16 +57,84 @@ void handleShutdown() {
 
-    std::string configName(osvr::server::getDefaultConfigFilename());
-    if (argc > 1) {
-        configName = argv[1];
+    std::string configName; // server configuration filename
+
+    opt::options_description optionsAll("All Options");
+    opt::options_description optionsVisible("Command Line Options");
+    opt::positional_options_description optionsPositional;
+
+    optionsPositional.add("config", -1);
+    optionsAll.add_options()(
+        "config",
+        opt::value<std::string>(&configName)
+            ->default_value(osvr::server::getDefaultConfigFilename()),
+        "server configuration filename");
+    optionsVisible.add_options()
+        ("help", "display this help message")
+        ("verbose,v", "enable verbose logging")
+        ("debug,d", "enable debug logging");
+    optionsAll.add(optionsVisible);
+
+    opt::variables_map values;
+    try {
+        opt::store(opt::command_line_parser(argc, argv)
+                       .options(optionsAll)
+                       .positional(optionsPositional)
+                       .run(),
+                   values);
+        opt::notify(values);
+    } catch (opt::invalid_command_line_syntax &e) {
+        log->error() << e.what();
+        return 1;
+    } catch (opt::unknown_option &e) {
+        log->error() << e.what(); // may want to replace boost msg
+        return 1;
+    }
+
+    if (values.count("help")) {
+        std::cout << optionsVisible << std::endl;
+        return 0;
+    }
+
+    if (values.count("debug")) {
+        osvr::util::log::LogRegistry::instance().setLevel(osvr::util::log::LogLevel::trace);
+        osvr::util::log::LogRegistry::instance().setConsoleLevel(osvr::util::log::LogLevel::trace);
+        log->trace("Debug logging enabled.");
+    } else if (values.count("verbose")) {
+        osvr::util::log::LogRegistry::instance().setLevel(osvr::util::log::LogLevel::debug);
+        osvr::util::log::LogRegistry::instance().setConsoleLevel(osvr::util::log::LogLevel::debug);
+        log->debug("Verbose logging enabled.");
+    }
+
+    configName = values["config"].as<std::string>();
+
+    boost::optional<fs::path> configPath(configName);
+    try {
+        if (!fs::exists(*configPath)) {
+            log->warn() << "File '" << configName
+                << "' not found.  Using empty configuration.";
+            configPath = boost::none;
+        } else {
+            if (fs::is_directory(*configPath)) {
+                log->error() << "'" << configName << "' is a directory.";
+                return -1;
+            } else if (!fs::is_regular_file(*configPath)) {
+                log->error() << "'" << configName << "' is special file.";
+                return -1;
+            }
+        }
+    } catch (fs::filesystem_error &e) {
+        log->error() << "Could not open config file at '" << configName << "'.";
+        log->error() << "Reason " << e.what() << ".";
+        configPath = boost::none;
+    }
+
+    if (configPath) {
```

Comment:
@mars979 I think so, since I would expect `fs::exists()` to return false on an empty string.